### PR TITLE
Consolidate AccountsDb constructors

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
Currently, the implementation for creating a new `AccountsDb` is spread across `AccountsDb::new_with_config()` and `AccountsDb::default_with_accounts_index()`. The implementation being fragmented duplicates some work/variable setting, and is a little hard to follow.

#### Summary of Changes
Roll the logic for `AccountsDb::default_with_accounts_index()` into `AccountsDb::new_with_config()`

### CODE REVIEW

File: accounts-db/src/accounts_db.rs [hunk 1]
```diff
@@ -1777,112 +1777,7 @@ impl AccountsDb {
...
-            partitioned_epoch_rewards_config: PartitionedEpochRewardsConfig::default(),
-            epoch_accounts_hash_manager: EpochAccountsHashManager::new_invalid(),
-            test_skip_rewrites_but_include_in_bank_hash: false,
-            latest_full_snapshot_slot: SeqLock::new(None),
-            is_experimental_accumulator_hash_enabled: default_accounts_db_config
-                .enable_experimental_accumulator_hash
-                .into(),
-            best_ancient_slots_to_shrink: RwLock::default(),
-        }
+        Self::new_single_for_tests()
```
Comment (steviez) [inline accounts-db/src/accounts_db.rs:1780]: I believe this substitution in `default_for_tests()` is valid; here are relevant functions that are getting called now:
https://github.com/anza-xyz/agave/blob/0e16e60e08fcfdb1b4a12f19ca4be4054a2707dc/accounts-db/src/accounts_db.rs#L1888-L1914

A few changes as far as I can tell:
- In the old code, `default_with_accounts_index()` would leave `.paths` and `.temp_paths` fields empty, whereas new code will create the temp dirs and corresponding paths + populate them
- The previous code used `AccountsDbConfig::default()` / defaults for fields whereas new code will use `ACCOUNTS_DB_CONFIG_FOR_TESTING`. This is still test code so I think this is fine too

I might lack some context / be missing something, but I'm currently under the impression that `AccountsDb::default_for_tests()` is redundant and can be removed. That seems like something for a subsequent PR tho

**Review by brooksprumo — COMMENTED** [review]
I think this looks right. Github doesn't make the diff particularly easy though... Are there any changes to the actual values used to create new instances of AccountsDb? I would imagine 'no', but want to double check.

**Comment by steviez** [comment]
> Are there any changes to the actual values used to create new instances of AccountsDb? I would imagine 'no', but want to double check.

No, the first commit was:
- Remove `..Self::default_with_accounts_index(...)` from `new_with_config()`
- Attempt to compile and see error that I'm missing ~30 fields; compiler lists out the missing fields
- Copy/paste those fields over from `default_with_accounts_index()`
- Remove `default_with_accounts_index()`

The subsequent changes were the less mechanical stuff

> Github doesn't make the diff particularly easy though...

Yeah, I see that. Not sure what I could have done differently; maybe break the move of fields into different commits, 5 fields per commit or something so they're more manageable ?

**Comment by brooksprumo** [comment]
> Not sure what I could have done differently; maybe break the move of fields into different commits, 5 fields per commit or something so they're more manageable ?

Nah, it's all good!

File: accounts-db/src/accounts_db.rs (hunk 1 above)
Comment (HaoranYi) [reply accounts-db/src/accounts_db.rs:1780]: yeah. we can remove `default_for_tests().

**Review by HaoranYi — APPROVED** [review]
lgtm.

**Review by brooksprumo — APPROVED** [review]

**Comment by steviez** [comment]
Did another manual review of the first commit (shifting of fields) and I believe those line up 1-to-1, so between that / your two reviews / green CI, I think we're good
